spec/services/sets/product_set_spec.rb

```diff
@@ -134,6 +134,27 @@
           end
         end
       end
+
+      context "when product attributes are not changed" do
+        let(:collection_hash) {
+          { 0 => { id: product.id, name: product.name } }
+        }
+
+        it 'returns true' do
+          is_expected.to eq true
+        end
+
+        it 'does not increase saved_count' do
+          subject
```

Comment:
It's a bit of a nit pick, but using `subject` here without knowing the context doesn't tell us what's happening. It'd rather be explicit and use `product_set.save` so I know straight away what we are testing.
Same for the various use of `subject` below.

Reply:
I haven't seen anywhere in the project style guide about it. So its comes down to personal preferences. In such cases i follow the [betterspecs.org](https://www.betterspecs.org/). If talking particularly about `subject` a lot arguments are here https://github.com/betterspecs/betterspecs/issues/7

I understand that exist different styles of writing tests. Another one is to put all context definition into into each `it`. Its described in book  "Testing Rails"(Josh Steiner, Joël Quenneville) https://books.thoughtbot.com/assets/testing-rails.pdf . But it does not looks like OFN follows this approach either.

>  but using subject here without knowing the context doesn't tell us what's happening

Sounds like a code smell. Why? Because `subject` describes what tested subject actually does. And its good when it does only one thing, because of Single Responsibility principle (SOLID). From my experience: If its not possible to write simple, obvious unit tests using rspec interface - code design is not optimal. Is the class `ProductSet` really that bad? Well, it uses inheritance, method override. include, extend, metaprogramming.... and it makes it sophistiated. But still its possible to make assumption that main(!) responsibility is to persist some data, and it is described here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/90c71c6a1a8391e5b7a2e24027bc1940834370f2/spec/services/sets/product_set_spec.rb#L5-L10
It follows rspec convention. So its good enough for me at the moment as for legacy code.

For context definition exists keyword `context`. Maybe it worth to extend description at this line https://github.com/openfoodfoundation/openfoodnetwork/pull/12515/files#diff-385c5ce991f97a247a8510a960775290559a04d5f0f4007219bdd346db00cc8eR138 to make it more obvious?

Reply:
or the spec can also look like:
```
expect {subject}.not_to change { product_set.saved_count }.from( 0 )
```

Reply:
Sorry I should have expressed myself better, it's more of a "personal nit pick". I am aware of the better spec recommendations, I just don't agree with all of it. But as you say it comes down to personal preference. 
Anyway I am happy for the code to be merged like this, the requested change was in relation to David's changes, which he updated. So all good now !